File: 2219-maximum-number-of-words-found-in-sentences/2219-maximum-number-of-words-found-in-sentences.cpp
class Solution {
public:
    int mostWordsFound(vector<string>& sentences) {
        int max_length=0;
        for(const auto &s : sentences){
            int length = count(s.begin(),s.end(),' ');
            max_length = max(max_length,length+1); 
        }
        return max_length;
    }
};

File: 2219-maximum-number-of-words-found-in-sentences/2219-maximum-number-of-words-found-in-sentences.java
class Solution {
    public int mostWordsFound(String[] sentences) {
        int[] result = new int[sentences.length];
        for(int i = 0; i < sentences.length; i++){
            result[i] = sentences[i].split("\\s").length;
        }
        return getmax(result,sentences.length);
    }
    static int getmax(int arr[], int n){
       if(n==1)
       return arr[0];

       return Math.max(arr[n-1], getmax(arr, n-1));
   }
}